crates/subspace-farmer/src/dsn.rs

```diff
@@ -0,0 +1,26 @@
+use std::ops::Range;
+
+use futures::Stream;
+use subspace_core_primitives::PieceIndexHash;
+
+use crate::PiecesToPlot;
```

Comment:
Please group all imports together, no need to split them with empty lines like this